### PR TITLE
Feature/dealer positions refactor

### LINKED ISSUE #1072: Dealer Refactor
Incorporate distance from desired position using the STPInfo's.
After CalculateInfo() do AssignRoles() and give the STPInfos to the dealer to use for distance calculation, probably in getScoreMatrix()

### PR DESCRIPTION
**Pull request by timovdk**
Refactored dealer to also use stpInfo data to make decisions on what robot should get what role.
`distribute`, `getScoreMatrix` and `getScoreForDistance` have new code, all other changes are formatting.
closes #1072

### CODE REVIEW

**Review by RobotJente — APPROVED** [review]

**Review by jordiluong — APPROVED** [review]

**Review by Ratoone — COMMENTED** [review]
Seems fine, only one problem. I will open a different issue for that (see #1080)

File: src/utilities/Dealer.cpp [hunk 1]
```diff
@@ -1,21 +1,28 @@
-#include <roboteam_utils/Hungarian.h>
+/**
+ * The dealer will check for the flags that are set in plays, but also for the distance
+ * to a position that a robot might need to travel to. The lower the score of a robot, the better.
+ * The robot with the lowest score for a role will get assigned to that role, unless it has already been assigned.
+ * In that case it will pick the second best and so on.
```
Comment (RobotJente) [inline src/utilities/Dealer.cpp]: I don't think this is how the Hungarian works but I could be wrong. This sounds quite greedy, I thought the hungarian tries to find a more global solution

Comment (timovdk) [reply src/utilities/Dealer.cpp]: Makes sense, my understanding of  the hungarian was wrong

Comment (timovdk) [inline src/utilities/Dealer.cpp]:
```suggestion
/**
 * The dealer will check for the flags that are set in plays, but also for the distance
 * to a position that a robot might need to travel to. The lower the score of a robot, the better.
```

**Review by RobotJente — APPROVED** [review]

**Pull request merged**